quinn-proto/src/varint.rs

```diff
@@ -150,7 +150,11 @@ impl Codec for VarInt {
             return Err(UnexpectedEnd);
         }
         let mut buf = [0; 8];
-        buf[0] = r.get_u8();
+        // This does not make use of `r.get_u8()` to avoid the additional bounds
+        // check. We already performed this one earlier on.
+        buf[0] = r.chunk()[0];
+        r.advance(1);
+
```

Comment:
Does this specific change actually help? I wouldn't expect this to have any difference in bounds checking semantics.

Reply:
There's an `assert!` in the `get_u8`, not just a rust array bounds check. I don't think those are removed. And since even without the "le" change this proved to be faster, I guess practical experience shows that it helps.